common/src/stack/command/stack/commands/load/json/plugin_host.py

```diff
@@ -0,0 +1,192 @@
+# @copyright@
+# Copyright (c) 2006 - 2018 Teradata
+# All rights reserved. Stacki(r) v5.x stacki.com
+# https://github.com/Teradata/stacki/blob/master/LICENSE.txt
+# @copyright@
+
+import stack.commands
+import json
+from stack.exception import CommandError
+
+class Plugin(stack.commands.Plugin, stack.commands.Command):
+	notifications = True
+
+	def provides(self):
+		return 'host'
+	def requires(self):
+		return [ 'software', 'environment', 'group', 'network', 'appliance', 'os', 'global' ]
+
+	def run(self, args):
+
+		# check if the user would like to import host data
+		# if there are no args, assume the user would like to import everthing
+		if args and 'host' not in args:
+			return
+
+		# self.owner.data contains the data from the json file defined in init
+		if 'host' in self.owner.data:
+			import_data = self.owner.data['host']
+		else:
+			self.owner.log.info('no host data in json file')
+			return
+
+		self.notify('\n\tLoading host\n')
+		# add each host then assign its various values to it
+		for host in import_data:
+			host_name = host['name']
+			parameters = [
+				host_name,
+				f'box={host["box"]}',
+				f'longname={host["appliancelongname"]}',
+				f'rack={host["rack"]}',
+				f'rank={host["rank"]}'
+			]
+			if host['environment']:
+				parameters.append(f'environment={host["environment"]}')
+			self.owner.try_command('add.host',parameters , f'adding host {host["name"]}', 'exists')
+
+
+			# iterate through each interface for the host and set it
+			for interface in host['interface']:
+				self.owner.try_command('add.host.interface',[ host_name, f'interface={interface["interface"]}'], f'adding interface {interface["interface"]}', 'exists')
+
+
+				# iterate over each key in interface, ignoring 'already exists' warnings
+			for interface in host['interface']:
+				for k, v in interface.items():
+					if v and k != 'interface' and k!= 'alias':
+						parameters = [
+							host_name,
+							f'{k}={v}',
+							f'interface={interface["interface"]}',
+							]
+						self.owner.try_command(f'set.host.interface.{k}', parameters, f'setting {host_name} interface {k}', 'exists')
+
+				# the alias cannot be set, so add it here. There can be multiple
+				for alias in interface['alias']:
+					parameters = [
+						host_name,
+						f'alias={alias["alias"]}',
+						f'interface={interface["interface"]}',
+						]
+					self.owner.try_command('add.host.alias', parameters, f'adding {host_name} alias {alias}', 'exists')
+
+			# iterate through each attr for the host and add it
+			for attr in host['attrs']:
+				attr_name = attr['name']
+				attr_value = attr['value']
+				if not isinstance(attr_value, str):
+					attr_value = ' '.join(attr['value'])
```

Comment:
I think I know the answer is 'no', but can attr_value ever be an int?  That would require a type conversion here.